core/sail/shacl/src/main/java/org/eclipse/rdf4j/sail/shacl/AST/AndPropertyShape.java

```diff
@@ -288,9 +290,22 @@ public String buildSparqlValidNodes(String targetVar) {
 
 	@Override
 	public Stream<StatementPattern> getStatementPatterns() {
-		return and
-				.stream()
+		StatementPattern subject = new StatementPattern(
+				new Var("?this"),
+				new Var(UUID.randomUUID().toString()),
+				new Var(UUID.randomUUID().toString())
+		);
+
+		StatementPattern object = new StatementPattern(
+				new Var(UUID.randomUUID().toString()),
```

Comment:
Are these randomly generated variable names ever serialized, for example as part of a SPARQL query? If so we'll need a more robust name assignement, because this will produce a variable name that won't be legal in SPARQL.

I'm pretty sure we have ready-made code somewhere (in the query parser utils perhaps) for this kind of thing. There should be a single place where we generate these kind of anonymous vars, so that if we need to change the name-generation logic for any reason, we don't have to change it all over the place.